internal/cfgplugins/staticnexthop.go

```diff
@@ -217,3 +230,112 @@ func NextHopGroupConfigForMulticloud(t *testing.T, dut *ondatra.DUTDevice, traff
 		configureNextHopGroups(t, ni, params)
 	}
 }
+
+// NextHopGroupConfigForIpOverUdp configures the interface next-hop-group config for ip over udp.
+func NextHopGroupConfigForIpOverUdp(t *testing.T, dut *ondatra.DUTDevice, ni *oc.NetworkInstance, params NexthopGroupUDPParams, deleteTtl bool) {
+	t.Helper()
+	if deviations.NextHopGroupOCUnsupported(dut) {
+		cli := ""
+		groupType := ""
+
+		switch dut.Vendor() {
+		case ondatra.ARISTA:
+			switch params.TrafficType {
+			case oc.Aft_EncapsulationHeaderType_UDPV4:
+				groupType = "ipv4-over-udp"
+			case oc.Aft_EncapsulationHeaderType_UDPV6:
+				groupType = "ipv6-over-udp"
+			}
+
+			if len(params.DstIp) > 0 {
+				tunnelDst := ""
+				for i, addr := range params.DstIp {
+					tunnelDst += fmt.Sprintf("entry %d tunnel-destination %s \n", i, addr)
+				}
+				cli = fmt.Sprintf(`
+					nexthop-group %s type %s
+					tunnel-source intf %s
+					fec hierarchical
+   					%s
+					`, params.NexthopGrpName, groupType, params.SrcIp, tunnelDst)
+				helpers.GnmiCLIConfig(t, dut, cli)
+			}
+			if params.TTL != 0 {
+				cli = fmt.Sprintf(`
+					nexthop-group %s type %s
+					ttl %v
+					`, params.NexthopGrpName, groupType, params.TTL)
+				helpers.GnmiCLIConfig(t, dut, cli)
+			}
+
+			if params.DSCP != 0 {
+				cli = fmt.Sprintf(`
+					nexthop-group %s type %s
+					tos %v
+					`, params.NexthopGrpName, groupType, params.DSCP)
+				helpers.GnmiCLIConfig(t, dut, cli)
+			}
+
+			if deleteTtl {
+				cli = fmt.Sprintf(
+					`nexthop-group %s type %s
+					no ttl %v
+					`, params.NexthopGrpName, groupType, params.TTL)
+				helpers.GnmiCLIConfig(t, dut, cli)
+			}
+
+			if params.DstUdpPort != 0 {
+				cli = fmt.Sprintf(`tunnel type %s udp destination port %v`, groupType, params.DstUdpPort)
+				helpers.GnmiCLIConfig(t, dut, cli)
+			}
+
+			newPolicyForwardingGueEncap(t, dut, params, "GUE-Policy")
+		default:
+			t.Logf("Unsupported vendor %s for native command support for deviation 'next-hop-group config'", dut.Vendor())
+		}
+	} else {
+		nhg := ni.GetOrCreateStatic().GetOrCreateNextHopGroup(params.NexthopGrpName)
+		nhg.GetOrCreateNextHop(params.Index).SetIndex(params.Index)
+
+		ueh1 := ni.GetOrCreateStatic().GetOrCreateNextHop(params.Index).GetOrCreateEncapHeader(1)
+		for _, addr := range params.DstIp {
+			ueh1.GetOrCreateUdpV4().SetDstIp(addr)
+		}
+		if params.TTL != 0 {
+			ueh1.GetOrCreateUdpV4().SetIpTtl(params.TTL)
+		}
+		ueh1.GetOrCreateUdpV4().SetSrcIp(params.SrcIp)
+		ueh1.GetOrCreateUdpV4().SetDscp(params.DSCP)
+		ueh1.GetOrCreateUdpV4().SetDstUdpPort(params.DstUdpPort)
+		ueh1.GetOrCreateUdpV4().SetSrcUdpPort(params.SrcUdpPort)
+	}
+
+}
+
+// newPolicyForwardingGueEncap configures policy forwading for gue encapsulation
+func newPolicyForwardingGueEncap(t *testing.T, dut *ondatra.DUTDevice, params NexthopGroupUDPParams, policyName string) {
```

Comment:
I think this function should probably be moved to staticroute.go because it's the arista deviation alternative to an OC static route when using a next-hop-group.  Probably the function could be named something like:

`staticRouteToNextHopGroupCLI` indicating that this function is specifically for CLI config which is used only for static routes needing to point to a next-hop-group. This way it is not confused with a static route that uses an  IP address as the next-hop.